lib/internal/streams/add-abort-signal.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const {
+  AbortError,
+  codes,
+} = require('internal/errors');
+
+const eos = require('internal/streams/end-of-stream');
+const { ERR_INVALID_ARG_TYPE } = codes;
+
+const {
+  validateAbortSignal,
+} = require('internal/validators');
```

Comment:
Can we avoid this? I prefer streams to be self-contained (readable-stream).

Reply:
We can inline the logic here this just does:

```js
const validateAbortSignal = hideStackFrames((signal, name) => {
  if (signal !== undefined &&
      (signal === null ||
       typeof signal !== 'object' ||
       !('aborted' in signal))) {
    throw new ERR_INVALID_ARG_TYPE(name, 'AbortSignal', signal);
  }
});
```

Would that be better?

Reply:
Yes, thank you. Please add a comment on why this is inlined.

Reply:
Pushed, squashed, added a comment